src/Parser.js

```diff
@@ -89,7 +89,7 @@ export default class Parser {
         // Use old \color behavior (same as LaTeX's \textcolor) if requested.
         // We do this after the macros object has been copied by MacroExpander.
         if (settings.colorIsTextColor) {
-            this.gullet.macros["\\color"] = "\\textcolor";
+            this.gullet.macros.set("\\color", "\\textcolor");
```

Comment:
Wouldn't this modify `settings.macro`?

Reply:
Indeed, good catch!  I've tweaked this to no longer happen, thanks to the new expression-level group, and added a corresponding test.